### PR TITLE
Network policy changes

### PR DESCRIPTION
**Pull request by leodotcloud**

### CODE REVIEW

**Comment by alena1108** [comment]
checkstyle errors should be fixed + make sure test_authorization passes for both v1 and v2 api

**Comment by alena1108** [comment]
We also need to add a python test. For create/update cases (test should cover all the fields including embedded)

**Comment by vincent99** [comment]
- [ ] `groupBy` should be an enum
- [ ] There should be validation to require exactly one of {`from && to`, `within`, `between`} on a given `networkPolicyRule` so you can't create invalid rules.

File: code/iaas/logic/src/main/java/io/cattle/platform/process/network/DefaultNetworkProcess.java [hunk 1]
```diff
@@ -61,6 +61,8 @@ protected void setDefaultNetwork(Long accountId) {
 
             if (CommonStatesConstants.ACTIVATING.equals(network.getState()) && newDefaultNetworkId == null) {
```
Comment (alena1108) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/network/DefaultNetworkProcess.java]: looks like lines 62 and 64 can be merged

File: code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/data/metadata/common/NetworkMetaData.java [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
+import io.cattle.platform.core.addon.NetworkPolicyRule;
+import io.cattle.platform.core.addon.NetworkPolicyRule.NetworkPolicyRuleAction;
+
 public class NetworkMetaData {
     String name;
     String uuid;
     boolean is_default;
     boolean host_ports;
     protected Map<String, Object> metadata;
+    NetworkPolicyRuleAction default_policy_action;
```
Comment (ibuildthecloud) [inline code/iaas/config-item/server/src/main/java/io/cattle/platform/configitem/context/data/metadata/common/NetworkMetaData.java]: Just change to string, and List<Object>.  Using actual types just add unneeded overhead in marshaller/unmarshaller I believe.

File: code/iaas/model/src/main/java/io/cattle/platform/core/addon/NetworkPolicyRule.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public enum NetworkPolicyRuleWithin {
+        stack, service, linked
+    }
+
+    NetworkPolicyRuleMember from, to, between;
+    NetworkPolicyRuleAction action;
+    NetworkPolicyRuleWithin within;
+    String[] ports;
```
Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/addon/NetworkPolicyRule.java]: what is the format for the port value? Given it is string, not integer

Comment (alena1108) [inline code/iaas/model/src/main/java/io/cattle/platform/core/addon/NetworkPolicyRule.java]: We should document it

File: tests/integration/cattletest/core/test_network_policy.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+def test_default_network_policy(super_client):
+    nd = super_client.wait_success(super_client.create_network_driver())
+    network = super_client.create_network(isPublic=True,
+                                          networkDriverId=nd.id,
+                                          defaultPolicyAction='deny',
+                                          policy=['{"within": "stack", "action": "allow"}'])
+    network = super_client.wait_success(network)
+    assert network.state == 'active'
+    assert network.defaultPolicyAction == 'deny'
+    assert len(network.policy) == 1
```
Comment (alena1108) [inline tests/integration/cattletest/core/test_network_policy.py]: can we also add a validation for network.policy fields? Just to make sure its every field/value was set

**Comment by alena1108** [comment]
Other than my minor comment on the test, LGTM. Assuming all the tests pass + we plan to add more validation handling in cattle later.

**Comment by leodotcloud** [comment]
@alena1108 I have a lot of tests for the validation of policy on the micro service side. Will add the ones in cattle later.

**Comment by leodotcloud** [comment]
@alena1108 @ibuildthecloud Once the PR is good to go, I would like to rebase, squash and resubmit the PR.

**Pull request merged**